wordpress-mobile/WordPress-iOS
# Network mocking for all UI and screenshot tests

This PR updates all of our UI and screenshot tests to use mocked networking using WireMock. The mocking functionality is shared with WPAndroid from the [WordPressMocks repo](https://github.com/wordpress-mobile/WordPressMocks).

It relies on the following open PRs:

- https://github.com/wordpress-mobile/WordPressMocks/pull/3

Other related PRs:

- https://github.com/wordpress-mobile/WordPress-iOS/pull/11655
- https://github.com/wordpress-mobile/WordPress-iOS/pull/11717
- https://github.com/wordpress-mobile/WordPress-iOS/pull/11762
- https://github.com/wordpress-mobile/WordPress-iOS/pull/11807
- https://github.com/wordpress-mobile/WordPress-Android/pull/9422
- https://github.com/wordpress-mobile/WordPressKit-iOS/pull/141
- https://github.com/wordpress-mobile/WordPressAuthenticator-iOS/pull/93
- https://github.com/wordpress-mobile/WordPressMocks/pull/1

To test:

- Check out this branch. Run `rake dependencies` as usual.
- Run `rake mocks` to start the mock server.
- Run the UI tests in the `WordPressUITests` target with Xcode.
- All tests should pass and there should be no errors/warnings in the terminal when you ran the mock server.

**Note:** The screenshot tests are broken in `develop` due to the recent stats changes. I think we should fix that in a separate change.

Update release notes:
